python_modules/dagster/dagster/config/config_type.py

```diff
@@ -15,6 +15,7 @@ class ConfigTypeKind(PythonEnum):
     SELECTOR = "SELECTOR"
     STRICT_SHAPE = "STRICT_SHAPE"
     PERMISSIVE_SHAPE = "PERMISSIVE_SHAPE"
+    MAP = "MAP"
```

Comment:
nit: can you round up array and noneable from below, feel like especially array should be near